Implement/communicator robot

DaanBoelhouwers opened the pull request:
Implement CommunicatorRobot, adds extra Error messages to Processor

TODO add Serial Library to virtual environment
(My IDE won't let me do this)

DaanBoelhouwers committed:
Create dictionaries for Arduino input and output

DaanBoelhouwers committed:
Implement rough version of _communicate

DaanBoelhouwers committed:
Merge branch 'master' of https://github.com/PbQuinn/DBL-Embedded-Systems-Group11 into implement/CommunicatorRobot

DaanBoelhouwers committed:
Fix some issues after pulling

DaanBoelhouwers committed:
Implement _communicate() in CommunicatorRobot

DaanBoelhouwers committed:
Implement (part of) initialize()

The function process_output is now public, since it is necessary
during the initialization phase.

DaanBoelhouwers committed:
Implement initialize, undo some previous changes

DaanBoelhouwers committed:
Improve initialize, add timeout timer

DaanBoelhouwers committed:
Add more detailed errors to initialize

Also add docstrings to some methods

DaanBoelhouwers committed:
Merge branch 'master' of https://github.com/PbQuinn/DBL-Embedded-Systems-Group11 into implement/CommunicatorRobot

DaanBoelhouwers committed:
Add error messages from Arduino to processor

DaanBoelhouwers committed:
Move string_disk call to secondary_color_detected

kasteelvuur reviewed (approved):
Everything should be correct once Adrian adds the serial library to the virtual environment

AdrianBeers committed:
Install pyserial, remove TODO

Installs pyserial and removes TODO.

kasteelvuur committed:
Replace stringer confirmation to tertiary_motion

kasteelvuur committed:
Update processor test cases

Fix string inconsistencies

DaanBoelhouwers committed:
Add separate Error Ping input

DaanBoelhouwers committed:
Add input processing for Error Ping

DaanBoelhouwers committed:
Update docstrings for processor

Also updated one docstring in communicator. Furthermore, even though I did
add them, I suggest we leave out the private methods in the
processor class docstring.
